source/common/config/grpc_mux_impl.h

```diff
@@ -96,6 +98,28 @@ class GrpcMuxImpl : public GrpcMux,
     TokenBucketPtr limit_request_;
     // Limits warning messages when too many requests is detected.
     TokenBucketPtr limit_log_;
+    // BackOffStrategy used for update rejected cases.
+    BackOffStrategyPtr rejected_backoff_strategy_;
+    // Timer used for update rejected cases.
+    Event::TimerPtr rejected_timer_;
+    // Was a DiscoveryRequest pending with rejected timer.
+    bool rejected_request_pending_{};
+
+    void enableRejectedTimer() {
+      // if rejected request is pending for this typrUrl, do not enable timer.
```

Comment:
@htuch Please validate if it is the right thing to do. I am not queuing further failures if one failure is pending